src/main/org/deidentifier/arx/ARXLattice.java

```diff
@@ -1196,6 +1202,36 @@ private void build(final Transformation optimum, Map<String, Integer> headermap)
         createRelationships(solutions, solutions.getTop().getIdentifier());
         createRelationships(solutions, solutions.getBottom().getIdentifier());
     }
+    
+    /**
+     * Build an ARX lattice for a solution space with only one element
+     * @param solutions 
+     * @param optimum
+     * @param headermap
+     */
+    private void buildSingle(SolutionSpace solutions, final Transformation optimum, Map<String, Integer> headermap) {
```

Comment:
Arent't we allowed to select any solution that we checked after executing the differentially private search process? SafePub deterministically selects the optimum. It might be OK, to allow users to manually select any pivot element, meaning that it is not necessary to restrict the lattice to only one element. However, I am unsure about this...

Reply:
In fact, it is not obvious if such a modification retains the expected differential privacy guarantees. In all relevant formulas/proofs, the search algorithm has the form of a function which returns a single generalization scheme. This modification would formally extend it to a function which returns a set of generalization schemes, and it would require further analytical investigation to determine possible effects on the privacy guarantees provided which could result from such an extension. Hence, at least for now, I would suggest to stay with the proposed solution which returns one generalization scheme and is formally safe.